src/app/util/ember-compatibility-functions.cpp

```diff
@@ -600,8 +600,13 @@ CHIP_ERROR WriteSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVReader & a
     return apWriteHandler->AddStatus(attributePathParams, imCode);
 }
 
-} // namespace app
-} // namespace chip
+void MatterReportingListAttributeChangeCallback(chip::EndpointId endpoint, chip::ClusterId clusterId, chip::AttributeId attributeId,
+                                                ListOperation listOp)
+{
+    // TODO: We need to trigger the reporting based on detailed list operations; delete/update/append...
+    // At this moment, we just mark the list attribute as dirty to send all fields in report.
```

Comment:
Oh, but how would a delete/update without an index help us scope this more narrowly?  That is, I don't understand how the API we're adding here will help us narrow down reporting in practice.

Reply:
This function is supposed to be called from platform when a list attribute managed by the framework changes, the platform does not maintain the index info right now.

I put TODO here, one possible solution is to add platform API to resolve the index from impacted entry and pass index info to MatterReportingListAttributeChangeCallback or resolve the index within MatterReportingListAttributeChangeCallback (looks challenging)

Reply:
Right, but my point is this: how would one use this API in the form this PR is introducing to do anything other than invalidate the whole list (which is what we do right now already)?   I can see handling Append specially, but not the other bits.  So to use any of the other bits we would need to change the API again, right?

Basically, I don't understand why we are introducing any enum values here other than "invalide the list" and "append".

Reply:
A list index needs to be passed in for delete/update operations.

As to the why we're adding these APIs that devolve ultimately to marking the entire list as changed, it's to ensure we at least have the right application side APIs settled and stable now that clusters can use, even if it devolves to a sub-optimal implementation today.

Later, when we have cycles, we can swap out that implementation to something that actually will handle deletes/updates correctly to deliver more optimal reports without actually changing consumers.

Reply:
Sure, I can understand the implementation always doing "mark the whole list changed".  I'm questioning the current API shape as presented in its PR and its ability to support an implementation that does anything other than "mark the whole list changed".